staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/converter.go

```diff
@@ -167,45 +170,137 @@ func (c *delegatingCRConverter) Convert(in, out, context interface{}) error {
 // UnstructuredList with the request's GV, populates it from storage, then calls conversion to convert
 // the individual items. This function assumes it never gets a v1.List.
 func (c *delegatingCRConverter) ConvertToVersion(in runtime.Object, target runtime.GroupVersioner) (runtime.Object, error) {
+	// Special-case typed scale conversion if this custom resource supports a scale endpoint
+	if c.convertScale {
+		if _, isInScale := in.(*autoscalingv1.Scale); isInScale {
+			return typedscheme.Scheme.ConvertToVersion(in, target)
+		}
+	}
+
 	fromGVK := in.GetObjectKind().GroupVersionKind()
 	toGVK, ok := target.KindForGroupVersionKinds([]schema.GroupVersionKind{fromGVK})
 	if !ok {
 		// TODO: should this be a typed error?
 		return nil, fmt.Errorf("%v is unstructured and is not suitable for converting to %q", fromGVK.String(), target)
 	}
 
+	isList := false
+	var list *unstructured.UnstructuredList
+	switch t := in.(type) {
+	case *unstructured.Unstructured:
+		list = &unstructured.UnstructuredList{Items: []unstructured.Unstructured{*t}}
+	case *unstructured.UnstructuredList:
+		list = t
+		isList = true
+	default:
+		return nil, fmt.Errorf("unexpected type %T", in)
+	}
+
```

Comment:
nit: break out into utility function, e.g. `toUnstructuredList(in runtime.Object) (list *unstructured.UnstructuredList, isList bool)` ?

Reply:
I'm okay with it.